File: TUnit.TestProject/LongFailures.cs
using TUnit.TestProject.Attributes;

namespace TUnit.TestProject;

[EngineTest(ExpectedResult.Failure)]
[Category("LongFailures")]
public class LongFailures
{
    private static int _counter;

    [Repeat(100)]
    [Test]
    public async Task LongFailure()
    {
        var timeProvider = TimeProviderContext.Current;
        await timeProvider.Delay(TimeSpan.FromMilliseconds(Interlocked.Increment(ref _counter)));
        throw new Exception($"Failure after {_counter} seconds");
    }
}

File: TUnit.TestProject.FSharp/LongFailures.fs
namespace TUnit.TestProject.FSharp

open System
open System.Threading
open System.Threading.Tasks
open TUnit.Core

[<Category("LongFailures")>]
type LongFailures() =
    static let mutable counter = 0

    [<Repeat(100)>]
    [<Test>]
    member _.LongFailure() : Task = task {
        let delay = Interlocked.Increment(&counter)
        do! Task.Delay(TimeSpan.FromSeconds(float delay))
        raise (Exception())
    }